pkg/router/controller/route_secret_manager_test.go

```diff
@@ -0,0 +1,1139 @@
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("other-sandbox", "tls-secret", corev1.SecretTypeTLS, map[string][]byte{}),
+			},
+			eventType: watch.Added,
+			allow:     true,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			expectedEventType: watch.Deleted,
+			expectedRejections: map[string]string{
+				"sandbox-route-test": "ExternalCertificateValidationFailed",
+			},
+			expectedError: true,
+		},
+		{
+			name: "route added with externalCertificate allowed but secret of incorrect type",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("sandbox", "tls-secret", corev1.SecretTypeBasicAuth, map[string][]byte{}),
+			},
+			eventType: watch.Added,
+			allow:     true,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			expectedEventType: watch.Deleted,
+			expectedRejections: map[string]string{
+				"sandbox-route-test": "ExternalCertificateValidationFailed",
+			},
+			expectedError: true,
+		},
+		{
+			name: "route added with externalCertificate allowed and correct secret",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("sandbox", "tls-secret", corev1.SecretTypeTLS, map[string][]byte{
+					"tls.crt": []byte("my-crt"),
+					"tls.key": []byte("my-key"),
+				}),
+			},
+			eventType: watch.Added,
+			allow:     true,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "my-crt",
+						Key:         "my-key",
+					},
+				},
+			},
+			expectedEventType:  watch.Added,
+			expectedRejections: map[string]string{},
+			expectedError:      false,
```

Comment:
This line can be removed, the default for boolean is `false`. Also, can you please move all the positive cases at the top or at the bottom? This would help the reader.

Reply:
>This line can be removed, the default for boolean is `false`. 

Done, thanks

>Also, can you please move all the positive cases at the top or at the bottom? This would help the reader.

Updated the logical grouping of the tests like the following:

- scenarios when route is added
- scenarios when route is updated (old route without externalCertificate, new route with externalCertificate)
- scenarios when route is updated (old route with externalCertificate, new route with externalCertificate)
- scenarios when route is updated (old route with externalCertificate, new route without externalCertificate)
- scenario when route is updated (old route without externalCertificate, new route without externalCertificate)
- scenarios when route is deleted

All the above logical groups now have all `negative` cases first, followed by all `positive` cases